www/js/app.js

```diff
@@ -2956,6 +2956,56 @@ function pushBrowserHistoryState (title, titleSearch) {
     }
     window.history.pushState(stateObj, stateLabel, urlParameters);
 }
+var dropup = document.getElementById('dropup');
+dropup.addEventListener('click', function () {
+    var ToCList = document.getElementById('ToCList');
+    ToCList.style.display = ToCList.style.display === 'block' ? 'none' : 'block';
```

Comment:
I think we're unnecessarily building the ToC every time the user presses the button, even if the click is supposed to hide the ToC. So, I suggest something like this:

```
    if (ToCList.style.display !== 'none') {
        ToCList.style.display = 'none';
    } else {
        ToCList.style.display = 'block';
        ... rest of code ...
    }
```

Reply:
Yeah, makes sense. I'll get it done tomorrow and add some comments.